### PR TITLE
Feature/ondemand durab

### PR DESCRIPTION
**Pull request by cezarfx**
Cloud only, added support for on-demand tables by using NosqlTable.capacityMode to NosqlCapacityMode.ON_DEMAND.
￼

### CODE REVIEW

**Review by connelly38 — APPROVED** [review]

File: src/main/java/com/oracle/nosql/spring/data/core/mapping/NosqlTable.java [hunk 1]
```diff
@@ -76,6 +103,13 @@
      */
     String consistency() default Constants.DEFAULT_TABLE_CONSISTENCY;
 
+    /**
+     * Sets the default durability for all write operations applied to this
+     * table. Valid values for this are defined in {@link Durability}.
+     * If not set the default value for this is COMMIT_NO_SYNC.
+     */
+    String durability() default Constants.DEFAULT_TABLE_CONSISTENCY;
```
Comment (connelly38) [inline src/main/java/com/oracle/nosql/spring/data/core/mapping/NosqlTable.java:111]: consistency? shouldn't this be DURABILITY?

File: pom.xml [hunk 2]
```diff
@@ -132,6 +132,14 @@
             <scope>compile</scope>
         </dependency>
 
+        <!-- To avoid javadoc warnings. -->
```
Comment (connelly38) [inline pom.xml:135]: Do we use findbugs in any other SDK? Has George approved this?

File: src/test/java/com/oracle/nosql/spring/data/test/app/AppConfig.java [hunk 3]
```diff
@@ -9,15 +9,31 @@
 import java.io.IOException;
 
 import com.oracle.nosql.spring.data.config.NosqlDbConfig;
+import com.oracle.nosql.spring.data.core.mapping.NosqlCapacityMode;
 import com.oracle.nosql.spring.data.repository.config.EnableNosqlRepositories;
 
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
```
Comment (connelly38) [inline src/test/java/com/oracle/nosql/spring/data/test/app/AppConfig.java:19]: Is there a Durability test?

File: pom.xml (hunk 2 above)
Comment (gmfeinberg) [reply pom.xml:135]: This kind of change needs to go through Mike B and how does it avoid javadoc warnings?

File: src/main/java/com/oracle/nosql/spring/data/core/mapping/NosqlTable.java (hunk 1 above)
Comment (cezarfx) [reply src/main/java/com/oracle/nosql/spring/data/core/mapping/NosqlTable.java]: Correct.

File: src/main/java/com/oracle/nosql/spring/data/core/mapping/NosqlTable.java [hunk 4]
```diff
@@ -76,6 +103,13 @@
      */
     String consistency() default Constants.DEFAULT_TABLE_CONSISTENCY;
 
+    /**
```
[the remark this answers is not in the document]
Comment (cezarfx) [reply src/main/java/com/oracle/nosql/spring/data/core/mapping/NosqlTable.java:114]: Added.

File: pom.xml (hunk 2 above)
Comment (cezarfx) [reply pom.xml]: I reverted this change. We seems one warning is not worth the trouble.

File: src/test/java/com/oracle/nosql/spring/data/test/app/AppConfig.java (hunk 3 above)
Comment (cezarfx) [reply src/test/java/com/oracle/nosql/spring/data/test/app/AppConfig.java:19]: added

**Comment by cezarfx** [comment]
Fixed the issues above and added a few more fixes and did some code reorg to fix and avoid future bugs.

**Review by connelly38 — APPROVED** [review]

**Pull request merged**